src/app.py

```diff
@@ -81,13 +83,16 @@ def store_data(json_data, csv_data):
 		raise
 
 
-def source_urls_to_pdfs(source_urls):
-	logging.info("Converting source websites into PDFs")
+def source_urls_to_pdfs(source_urls, folder, names=None):
+	logging.info("Converting websites into PDFs")
 	pdfs = []
-	for source_url in source_urls:
-		parsed_url = urlparse(source_url)
-		name = f"{parsed_url.path.replace('/', '_')[1:]}.pdf"
-		if bucket_contains(name):
+	if not names:
+		names = [f"{urlparse(source_url).path.replace('/', '_')[1:]}.pdf" for source_url in source_urls]
+	else:
+		assert len(names) == len(source_urls)
+    names = [n + ".pdf" if ".pdf" not in n for n in names]  # ensure .pdf suffix
```

Comment:
If the URL already points to a PDF, we should just download it (probably using `requests`). Using `pdfkit` will convert HTML to PDF, which might mangle some of the downloads (or leave them blank).